DigitalLearningSolutions.Web/Controllers/Register/RegisterAdminController.cs

```diff
@@ -0,0 +1,250 @@
+            return View(new PasswordViewModel());
+        }
+
+        [ServiceFilter(typeof(RedirectEmptySessionData<RegistrationData>))]
+        [HttpPost]
+        public IActionResult Password(PasswordViewModel model)
+        {
+            if (!ModelState.IsValid)
+            {
+                return View(model);
+            }
+
+            var data = TempData.Peek<RegistrationData>()!;
+            data.PasswordHash = cryptoService.GetPasswordHash(model.Password!);
+            TempData.Set(data);
+
+            return RedirectToAction("Summary");
+        }
+
+        [ServiceFilter(typeof(RedirectEmptySessionData<RegistrationData>))]
+        [HttpGet]
+        public IActionResult Summary()
+        {
+            var data = TempData.Peek<RegistrationData>()!;
+            var model = new SummaryViewModel(data);
+            PopulateSummaryExtraFields(model, data);
+            return View(model);
+        }
+
+        [ServiceFilter(typeof(RedirectEmptySessionData<RegistrationData>))]
+        [HttpPost]
+        public IActionResult Summary(SummaryViewModel model)
+        {
+            var data = TempData.Peek<RegistrationData>()!;
+
+            if (!ModelState.IsValid)
+            {
+                var viewModel = new SummaryViewModel(data);
+                PopulateSummaryExtraFields(viewModel, data);
+                viewModel.Terms = model.Terms;
+                return View(viewModel);
+            }
+
+            // TODO: (HEEDLS-527) register admin details and notification preferences in database
+
+            return RedirectToAction("Confirmation");
+        }
+
+        [HttpGet]
+        public IActionResult Confirmation()
+        {
+            return View();
+        }
+
+        private bool IsRegisterAdminAllowed(int centreId)
+        {
+            if (centresDataService.GetCentreName(centreId) == null)
+            {
+                return false;
+            }
+
+            var (autoRegistered, autoRegisterManagerEmail) = centresDataService.GetCentreAutoRegisterValues(centreId);
+            return !autoRegistered && !string.IsNullOrWhiteSpace(autoRegisterManagerEmail);
+        }
+
+        private void SetAdminRegistrationData(int centreId)
+        {
+            var adminRegistrationData = new RegistrationData(centreId);
+            var id = adminRegistrationData.Id;
+
+            Response.Cookies.Append(
+                CookieName,
+                id.ToString(),
+                new CookieOptions
+                {
+                    Expires = DateTimeOffset.UtcNow.AddDays(30)
+                }
+            );
+
+            TempData.Clear();
+            TempData.Set(adminRegistrationData);
+        }
+
+        private void ValidateEmailAddress(string? email, int centreId)
+        {
+            if (email == null)
+            {
+                return;
+            }
+
+            var (_, autoRegisterManagerEmail) = centresDataService.GetCentreAutoRegisterValues(centreId);

```

Comment:
Super minor, but you should be able to do:
`var autoRegisterManagerEmail = centresDataService.GetCentreAutoRegisterValues(centreId).autoRegisterManagerEmail;`